packages/components/src/utils/time.ts

```diff
@@ -202,12 +202,20 @@ export function getLocalizedDecimalSeparator(locale: Locale, numberingSystem: Nu
   return numberStringFormatter.localize("1.1").split("")[1];
 }
 
-export function getLocalizedTimePartSuffix(
-  part: "hour" | "minute" | "second",
-  locale: Locale,
-  numberingSystem: NumberingSystem = "latn",
-): string {
-  const formatter = createLocaleDateTimeFormatter({ locale, numberingSystem });
+export function getLocalizedTimePartSuffix({
+  hour12,
+  locale,
+  numberingSystem = "latn",
+  part,
+  step,
+}: {
+  hour12: boolean;
+  locale: Locale;
+  numberingSystem: NumberingSystem;
+  part: Extract<"hour" | "minute" | "second", TimePart>;
```

Comment:
The first argument for [`Extract`](https://www.typescriptlang.org/docs/handbook/utility-types.html#extracttype-union) is the type to extract from, so this should be:

```ts
Extract<TimePart, "hour" | "minute" | "second">;
```